hummingbot/market/bittrex/bittrex_auth.py

```diff
@@ -0,0 +1,63 @@
+import time
+import hmac
+import hashlib
+from typing import Dict, Any
+
+import ujson
+
+
+class BittrexAuth:
+    def __init__(self, api_key: str, secret_key: str):
+        self.api_key = api_key
+        self.secret_key = secret_key
+
+    def generate_auth_dict(
+        self,
+        http_method: str,
+        url: str,
+        params: Dict[str, Any] = None,
+        body: Dict[str, Any] = None,
+        subaccount_id: str = "",
+    ) -> Dict[str, any]:
+        """
+        Generates the url and the valid signature to authenticate with the API endpoint.
+        :param http_method: String representing the HTTP method in use ['GET', 'POST', 'DELETE'].
+        :param url: String representing the API endpoint.
+        :param params: Dictionary of url parameters to be included in the api request. USED ONLY IN SOME CASES
+        :param body: Dictionary representing the values in a request body.
+        :param subaccount_id: String value of subaccount id.
+        :return: Dictionary containing the final 'params' and its corresponding 'signature'.
+        """
+
+        # Appends params the url
+        def append_params_to_url(url: str, params: Dict[str, any] = {}) -> str:
+            if params:
+                param_str = "&".join([f"{param}={value}" for (param, value) in params.items()])
+                return f"{url}?{param_str}"
+            return url
+
+        def construct_content_hash(body: Dict[str, any] = {}) -> str:
+            if body:
+                return hashlib.sha512(ujson.dumps(body).encode()).hexdigest()
```

Comment:
You should preferably only do the JSON encode() once - and make sure the JSON you send out and the the JSON you signed are exactly the same. While doing it twice "should" return the same result, it's both more risky and performs unnecessary computations.

Reply:
The reason for performing `encode()` here is because its must be encoded before hashing. If not, `hashlib` would throw a `TypeError`

Reply:
No, I mean you can do it like this..

```
def construct_content_hash(body: bytes) -> str:
   ...

...
json_bytes: bytes = ujson.dumps(body).encode()
content_hash = construct_content_hash(json_bytes)
...
return {"headers": headers, "body": json_bytes, "url": url}
```

The encode I mentioned meant converting from Dict to JSON bytes. If you only do the conversion once, then you can save